keps/sig-cloud-provider/2699-add-webhook-hosting-to-ccm/README.md

```diff
@@ -0,0 +1,455 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP will detail enhancing the CCM framework to support cloud provider
+specific webhooks. The intent is to make it easy to either generate a binary
+or enhance the existing CCM binary to host such webhooks. We also intend to
+allow for easily linking in "standard" webhooks needed by other SIGs which
+need to be customized for particular cloud providers.
+
+## Motivation
+
+The Cloud (Provider) Controller Manager (CCM) is the binary into which the Cloud
+Provider places all the controllers needed to make a Kubernetes cluster work
+correctly on their Cloud. There are also occasions when it makes sense for a Cloud
+Provider to want these customizations to be applied inline rather than 
+asynchronously after a change has already been applied. 
+
+Our initial example of this is from SIG Storage. These would like the functionality 
+from PVL admission controller (https://github.com/kubernetes/kubernetes/issues/52617).
+This needs to be completed for cloud provider extraction to complete. Several
+Cloud Providers have indicated that this should be done inline, especially as the
+existing deprecated solution is an inline solution.
+
+### Goals
+
+Our current in tree Cloud Providers are able to stop using the existing PVL admission 
```

Comment:
The way this KEP is written it seems like the goal would be more generic, i.e. to have a framework that allows cloud providers to easily run webhooks in their cloud-controller-manager, but if this is the goal, then I think more information about the PVL scenario should be added to the motivation and user stories sections.  And to be clear I think its probably better to make this KEP more about PVL, because its the primary driver for this feature.

Reply:
Will add a bit more but I don't want to sign up for writing the PVL webhook. Happy to help wire it in but I'm not sure we should be writing the actual PVL webhook.

Reply:
That's totally fair but we should have a KEP for the PVL webhook to reference in this KEP, since its the driving reason behind this KEP, imo. Otherwise its hard to explain why we definitely need this.  Are the reasons for a webhook and not a controller documented in detail somewhere?

Reply:
cc @msau42